src/vp8/util/generic_worker.cc

```diff
@@ -36,6 +40,8 @@
 void _cross_platform_pause() {
 #if !defined(USE_SCALAR) && defined(__i386__)
         _mm_pause();
+#elif __ARM_ACLE
+        __yield();
```

Comment:
Please do not use yield, that is implemented as a nop, and so it takes no cycles to execute as it is discarded by the CPU frontend.  That will not have the intended effect of delaying the execution of the thread by some amount of time.

Instead of `__yield();` please use `__asm__ __volatile__("dmb ish");`
`dmb` is a Data Memory Barrier which provides a small delay of the order of several dozens cycles.

Reply:
Docs say it's a hint, just like _mm_pause:

https://developer.arm.com/documentation/ddi0596/2021-03/Base-Instructions/YIELD--YIELD-?lang=en
https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=mm_pause&ig_expand=5201

Looking at your profile, though, I guess I should follow your advice.
Would you say, that this is Graviton specific?

Reply:
All the Arm cpus we've looked at pause works better as an `isb`. `yield` is a nop and doesn't actually help in a spin loop.